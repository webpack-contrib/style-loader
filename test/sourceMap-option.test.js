import compile from './helpers/compiler';
import runTestInJsdom from './helpers/runTestInJsdom';

describe('sourceMap option', () => {
  it('not specified', async () => {
    expect.assertions(3);

    const testId = './simple.js';
    const stats = await compile(testId, {
      devtool: 'source-map',
    });

    runTestInJsdom(stats, (dom) => {
      expect(dom.serialize()).toMatchSnapshot('DOM');
    });

    expect(stats.compilation.warnings).toMatchSnapshot('warnings');
    expect(stats.compilation.errors).toMatchSnapshot('errors');
  });

  it('false', async () => {
    expect.assertions(3);

    const testId = './simple.js';
    const stats = await compile(testId, {
      devtool: 'source-map',
      loader: {
        options: {
          sourceMap: false,
        },
      },
    });

    runTestInJsdom(stats, (dom) => {
      expect(dom.serialize()).toMatchSnapshot('DOM');
    });

    expect(stats.compilation.warnings).toMatchSnapshot('warnings');
    expect(stats.compilation.errors).toMatchSnapshot('errors');
  });

  it('true', async () => {
    expect.assertions(3);

    const testId = './simple.js';
    const stats = await compile(testId, {
      devtool: 'source-map',
      loader: {
        options: {
          sourceMap: true,
        },
      },
      cssLoader: {
        options: {
          sourceMap: true,
        },
      },
    });

    runTestInJsdom(stats, (dom) => {
      expect(dom.serialize()).toMatchSnapshot('DOM');
    });

    expect(stats.compilation.warnings).toMatchSnapshot('warnings');
    expect(stats.compilation.errors).toMatchSnapshot('errors');
  });

  it('true, but previous loader do not generate source map', async () => {
    expect.assertions(3);

    const testId = './simple.js';
    const stats = await compile(testId, {
      devtool: 'source-map',
      loader: {
        options: {
          sourceMap: true,
        },
      },
      cssLoader: {
        options: {
          sourceMap: false,
        },
      },
    });

    runTestInJsdom(stats, (dom) => {
      expect(dom.serialize()).toMatchSnapshot('DOM');
    });

    expect(stats.compilation.warnings).toMatchSnapshot('warnings');
    expect(stats.compilation.errors).toMatchSnapshot('errors');
  });
});
