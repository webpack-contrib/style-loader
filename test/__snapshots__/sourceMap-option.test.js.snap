// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sourceMap option false: DOM 1`] = `
"<!DOCTYPE html><html><head>
  <title>style-loader test</title>
  <style id=\\"existing-style\\">.existing { color: yellow }</style>
<style type=\\"text/css\\">body {
  color: red;
}
</style></head>
<body>
  <h1>Body</h1>
  <div class=\\"target\\"></div>
  <iframe class=\\"iframeTarget\\"></iframe>


</body></html>"
`;

exports[`sourceMap option false: errors 1`] = `Array []`;

exports[`sourceMap option false: warnings 1`] = `Array []`;

exports[`sourceMap option not specified: DOM 1`] = `
"<!DOCTYPE html><html><head>
  <title>style-loader test</title>
  <style id=\\"existing-style\\">.existing { color: yellow }</style>
<style type=\\"text/css\\">body {
  color: red;
}
</style></head>
<body>
  <h1>Body</h1>
  <div class=\\"target\\"></div>
  <iframe class=\\"iframeTarget\\"></iframe>


</body></html>"
`;

exports[`sourceMap option not specified: errors 1`] = `Array []`;

exports[`sourceMap option not specified: warnings 1`] = `Array []`;

exports[`sourceMap option true, but previous loader do not generate source map: DOM 1`] = `
"<!DOCTYPE html><html><head>
  <title>style-loader test</title>
  <style id=\\"existing-style\\">.existing { color: yellow }</style>
<style type=\\"text/css\\">body {
  color: red;
}
</style></head>
<body>
  <h1>Body</h1>
  <div class=\\"target\\"></div>
  <iframe class=\\"iframeTarget\\"></iframe>


</body></html>"
`;

exports[`sourceMap option true, but previous loader do not generate source map: errors 1`] = `Array []`;

exports[`sourceMap option true, but previous loader do not generate source map: warnings 1`] = `Array []`;

exports[`sourceMap option true: DOM 1`] = `
"<!DOCTYPE html><html><head>
  <title>style-loader test</title>
  <style id=\\"existing-style\\">.existing { color: yellow }</style>
<style type=\\"text/css\\">body {
  color: red;
}

/*# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbInN0eWxlLmNzcyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFBQTtFQUNFLFVBQVU7QUFDWiIsImZpbGUiOiJzdHlsZS5jc3MiLCJzb3VyY2VzQ29udGVudCI6WyJib2R5IHtcbiAgY29sb3I6IHJlZDtcbn1cbiJdfQ== */</style></head>
<body>
  <h1>Body</h1>
  <div class=\\"target\\"></div>
  <iframe class=\\"iframeTarget\\"></iframe>


</body></html>"
`;

exports[`sourceMap option true: errors 1`] = `Array []`;

exports[`sourceMap option true: warnings 1`] = `Array []`;
